src/m365/spo/commands/folder/folder-retentionlabel-remove.ts

```diff
@@ -0,0 +1,158 @@
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.folderId &&
+          !validation.isValidGuid(args.options.folderId as string)) {
+          return `${args.options.folderId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['folderUrl', 'folderId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (args.options.confirm) {
+      await this.removeFolderRetentionLabel(logger, args);
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to remove the retentionlabel from folder ${args.options.folderId || args.options.folderUrl} located in site ${args.options.webUrl}?`
+      });
+
+      if (result.continue) {
+        await this.removeFolderRetentionLabel(logger, args);
+      }
+    }
+  }
+
+  private async removeFolderRetentionLabel(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Removing retention label from folder ${args.options.folderId || args.options.folderUrl} in site at ${args.options.webUrl}...`);
+    }
+    try {
+      const folderProperties = await this.getFolderProperties(args);
+      const options: SpoListItemRetentionLabelRemoveCommandOptions = {
+        webUrl: args.options.webUrl,
+        listUrl: folderProperties.listServerRelativeUrl,
+        listItemId: folderProperties.id,
+        confirm: true,
+        output: 'json',
+        debug: this.debug,
+        verbose: this.verbose
+      };
+
+      await Cli.executeCommandWithOutput(SpoListItemRetentionLabelRemoveCommand as Command, { options: { ...options, _: [] } });
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getFolderProperties(args: CommandArgs): Promise<{ id: string, listServerRelativeUrl: string }> {
+    const requestOptions: AxiosRequestConfig = {
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
+
+    if (args.options.folderId) {
+      requestOptions.url = `${args.options.webUrl}/_api/web/GetFolderById('${args.options.folderId}')?$expand=ListItemAllFields`;
+    }
+    else {
+      requestOptions.url = `${args.options.webUrl}/_api/web/GetFolderByServerRelativeUrl('${formatting.encodeQueryParameter(args.options.folderUrl!)}')?$expand=ListItemAllFields`;
+    }
+
+    const response = await request.get<FolderProperties>(requestOptions);
+    return { id: response.ListItemAllFields.Id, listServerRelativeUrl: this.getListServerRelativeUrl(response.ServerRelativeUrl) };
+  }
+
+  private getListServerRelativeUrl(folderUrl: string): string {
+    return folderUrl.replace(/\/[^\/]+$/, '');
```

Comment:
This does not get you the `listServerRelativeUrl` perse, what if I execute the command on a subfolder? (/sites/sitename/library/folder/subfolder) 

You should call the endpoint slightly different to include the right information:

`https://contoso.sharepoint.com/sites/sales/_api/Web/GetFolderById(guid'e1b6d261-b54a-4202-b486-6ed6d0685d68')?$expand=ListItemAllFields,ListItemAllFields/ParentList&$select=ListItemAllFields/ParentList/Id,ListItemAllFields/Id`

This will get you the list item id and list id.

Reply:
Great feedback 😍, thank you!